vaadin-login-flow-parent/vaadin-login-flow/src/main/java/com/vaadin/flow/component/login/LoginOverlay.java

```diff
@@ -242,6 +242,10 @@ public String getDescription() {
      * overlay custom form area. This area is displayed only if there's at least
      * one component added with {@link LoginOverlayContent#add(Component...)}.
      *
+     * Fields that are part of custom form area are not automatically submitted
+     * as part of the {@link LoginForm.LoginEvent} and their values need to be
+     * synced using separate events, which interferes with the POST request.
```

Comment:
This mixes several issues which makes understanding this confusing:
- Custom fields are not part of the `LoginEvent`. We can definitely mention that here, and that you need to access their values separately from the event.
- The POST request itself is not related to the `LoginEvent`. When you set an `action` the form just does a POST to whatever URL you set there (so some Spring security handler probably), but you don't handle the login event itself in that case.
- From testing this I can at least see that I get JS errors if Flow does a sync request for the custom field while the browser does the login request in parallel. The other thing is that the custom field is not part of the form, so whatever handles the login request can not use it. So from a practical perspective, the second thing to mention is that custom fields are straight up not supported when setting an `action` on the LoginOverlay, as their values will not be part of the login request.

Reply:
Thank you for the note. Updated the JavaDoc to also include the following:

>  not supported when setting an action on the LoginOverlay, as their values will not be part of the login request.

Reply:
> From testing this I can at least see that I get JS errors if Flow does a sync request for the custom field while the browser does the login request in parallel. 

Is that the same as reported in https://github.com/vaadin/flow-components/issues/6098#issuecomment-1980190986? I think this is the actual issue that needs to be fixed.

Reply:
> Is that the same as reported in https://github.com/vaadin/flow-components/issues/6098#issuecomment-1980190986? I think this is the actual issue that needs to be fixed.

I'm not sure, the issue description is quite vague in what "interferes with the POST request" means, and whether the login effectively works or not. For me it was working, there was just an error to go with it.